### PR TITLE
drivers: mipi: delay initialization after display drivers

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Currently MIPI devices are set to initialize before display drivers, which is incoherent with how the mcux mipi-dsi device is declared in the device tree.

This is detected with build time priority checking:

west build -p -b mimxrt595_evk_cm33 samples/drivers/display \
	     -DCONFIG_CHECK_INIT_PRIORITIES=y

ERROR: /soc/peripheral@50000000/mipi_dsi@31000 POST_KERNEL 40 < \
		/soc/peripheral@50000000/lcdif@210000 POST_KERNEL 85

Note that this also changes the priority of dsi_stm32, though that should be ok since does not appear to have other dependencies in devicetree.

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
@danieldegrasse @erwango hi, one more priority change to match priorities with the devicetree hirearchy, could you test this for me? It impacts both `nxp,imx-mipi-dsi` and `st,stm32-mipi-dsi`

**Comment by fabiobaltieri** [comment]
Wow, no maintainers :-(

**Review by cfriedt — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]
Works for me on an RT595 and RT1170 EVK.

**Comment by cfriedt** [comment]
Sorry - I just saw that we were maybe waiting on feedback from @erwango.

**Comment by fabiobaltieri** [comment]
> Sorry - I just saw that we were maybe waiting on feedback from @erwango.

Yeah actually now that I went through the combination for finding the affected board-shield I realize this may have broke it:

```
$ west build -p -b stm32h747i_disco_m7 samples/subsys/display/lvgl -DSHIELD=st_b_lcd40_dsi1_mb1166 -DCONFIG_CHECK_INIT_PRIORITIES=y
...
WARNING: /soc/dsihost@50000000/otm8009a@0 POST_KERNEL 86 == /soc/dsihost@50000000 POST_KERNEL 86
```

Sorry about it, should have dug this out before sending this PR, will send a followup to bump that one up, it's in the board specific config anyway.

**Comment by fabiobaltieri** [comment]
There you go: https://github.com/zephyrproject-rtos/zephyr/pull/60178, should be stable with that.

**Pull request merged**
